### PR TITLE
Move parameter typing for extraSettings

### PR DESCRIPTION
**Pull request by Chartman123**

### CODE REVIEW

File: lib/Db/Question.php [hunk 1]
```diff
@@ -79,8 +79,6 @@ public function getExtraSettings(): array {
 	 * @param object|array $extraSettings
 	 */
 	public function setExtraSettings($extraSettings) {
```
Comment (susnux) [inline lib/Db/Question.php:81]:
```suggestion
	public function setExtraSettings(array $extraSettings) {
```

I think the todo meant this

Comment (Chartman123) [reply lib/Db/Question.php:81]: Ah ok, I thought it was about the param description above...

Comment (susnux) [reply lib/Db/Question.php:81]: No php 8 allowed typings like `public function foo(array|object $bar)` for older PHP versions you needed that comment

Comment (Chartman123) [reply lib/Db/Question.php:81]: So should we keep the `array|object` typing? And can we remove the comment, too?

File: lib/Db/Question.php [hunk 2]
```diff
@@ -75,12 +75,7 @@ public function getExtraSettings(): array {
 		return json_decode($this->getExtraSettingsJson() ?: '{}', true); // assoc=true, => Convert to associative Array
 	}
 
-	/**
-	 * @param object|array $extraSettings
-	 */
-	public function setExtraSettings($extraSettings) {
-		// TODO: When the php requirement is >= 8.0 change parameter typing to `object|array` to allow assoc. arrays from `Question::fromParams`
-		
+	public function setExtraSettings(object|array $extraSettings) {
```
Comment (susnux) [inline lib/Db/Question.php:78]:
```suggestion
	public function setExtraSettings(array $extraSettings) {
```

Is it anytime an object? I think this is a leftover from the object -> associative array migration

Comment (Chartman123) [reply lib/Db/Question.php:78]:
```
{
  "id":87,
  "keyValuePairs":{
    "extraSettings":{
      "shuffleOptions":true,
      "allowOtherAnswer":false
    }
  }
}
```
This is at least what we send from the frontend on updating the extraSettings of a question...

Comment (susnux) [reply lib/Db/Question.php:78]: Yes but I normally JS objects should be encoded as associative arrays on PHP.

Comment (Chartman123) [reply lib/Db/Question.php:78]: @susnux yes, you're right: `Array ( [extraSettings] => Array ( [shuffleOptions] => 1 [allowOtherAnswer] => 1) ) `

**Review by susnux — APPROVED** [review]

**Pull request merged**